src/Illuminate/Database/Query/Builder.php

```diff
@@ -2275,7 +2275,7 @@ public function toSqlWithBindings()
             '?',
             collect($this->connection->prepareBindings($this->getBindings()))
                 ->map(function ($i) {
-                    return (is_string($i)) ? "'{$i}'" : $i;
+                    return (is_string($i)) ? $this->connection->getPdo()->quote($i) : $i;
```

Comment:
`$this->connection` is type-hinted as a `Illuminate\Database\ConnectionInterface`, which does not have a `getPdo()` method.

Reply:
@rodrigopedra Do you think something like this would be better?
```php
$search = ["\\",  "\x00", "\n",  "\r",  "'",  '"', "\x1a"];
$replace = ["\\\\","\\0","\\n", "\\r", "\'", '\"', "\\Z"];

return "'" . str_replace($search, $replace, $value) . "'";
```

Reply:
I think a combination of both, similar to what debugbar does:

https://github.com/barryvdh/laravel-debugbar/blob/6420113d90bb746423fa70b9940e9e7c26ebc121/src/DataCollector/QueryCollector.php#L121-L126

https://github.com/barryvdh/laravel-debugbar/blob/6420113d90bb746423fa70b9940e9e7c26ebc121/src/DataCollector/QueryCollector.php#L148-L152

https://github.com/barryvdh/laravel-debugbar/blob/6420113d90bb746423fa70b9940e9e7c26ebc121/src/DataCollector/QueryCollector.php#L191-L197

Although I would use `method_exists` instead of `try...catch` to check for PDO

Reply:
@rodrigopedra done.